crates/starknet-devnet-types/src/rpc/transactions/l1_handler_transaction.rs

```diff
@@ -110,8 +111,33 @@ impl L1HandlerTransaction {
     }
 }
 
+impl TryFrom<&L1HandlerTransaction> for MessageToL2 {
```

Comment:
Why not simply implement `From`?

Reply:
because of this:
 ```rust
let l1_contract_address = value.calldata.get(0).ok_or(Error::ConversionError(
            ConversionError::InvalidInternalStructure(
                "L1HandlerTransaction calldata is expected to have at least one element"
                    .to_string(),
            ),
        ))?;```
and this:
```rust
ContractAddress::new(*l1_contract_address)?
```

Reply:
Sure you can't use question marks, but you can do unwrap inside of the `From` implementation.

Reply:
Yes, but we try to avoid using unwraps, due to the possibility of panicking

Reply:
Ok, sorry for the confusion. Will resolve and approve.